packages/cmrjs/cmr-utils.js

```diff
@@ -476,6 +497,50 @@ function getCreds() {
   };
 }
 
+function generateEcho10XMLString(granule) {
+  const mapping = new Map([]);
+  Object.keys(granule).forEach((key) => {
+    if (key === 'OnlineAccessURLs') {
+      mapping.set(key, granule[key]);
+      mapping.set('OnlineResources', granule.OnlineResources);
+    }
+    else if (key !== 'OnlineResources') {
+      mapping.set(key, granule[key]);
+    }
+  });
+  return js2xmlParser.parse('Granule', mapping);
+}
+/**
+ * Updates CMR xml file with 'xml' string
+ *
+ * @param  {string} xml - XML to write to cmrFile
+ * @param  {Object} cmrFile - cmr file object to write xml to
+ * @returns {Promise} returns promised aws.promiseS3Upload response
+ */
+async function uploadEcho10CMRFile(xml, cmrFile) {
+  const tags = await aws.s3GetObjectTagging(cmrFile.bucket, getS3KeyOfFile(cmrFile));
+  const tagsQueryString = aws.s3TagSetToQueryString(tags.TagSet);
+  return aws.promiseS3Upload({
+    Bucket: cmrFile.bucket, Key: getS3KeyOfFile(cmrFile), Body: xml, Tagging: tagsQueryString
+  });
+}
+/**
+ * Method takes an array of URL objects to update, an 'origin' array of original URLs
+ * and a list of URLs to remove and returns an array of merged URL objectgs
+ *
+ * @param  {Array<Object>} URLlist - array of URL objects
+ * @param  {Array<Object>} originalURLlist - array of URL objects
+ * @param  {Array<Object>} removedURLs - array of URL objects
+ * @param  {Array<Object>} URLTypes - array of UMM/Echo FileTypes to include
+ * @param  {Array<Object>} URLlistFieldFilter - array of URL Object keys to omit
+ * @returns {Array<Object>} array of merged URL objects, filtered
+ */
+function buildMergedEchoURLObject(URLlist = [], originalURLlist = [], removedURLs = [],
+  URLTypes, URLlistFieldFilter) {
+  let filteredURLObjectList = URLlist.filter((urlObj) => URLTypes.includes(urlObj.Type));
```

Comment:
With re-assignment this function is a little difficult for me to read. What do you think about:
```suggestion
const includesType = (urlObj) => URLTypes.includes(urlObj.Type);
const omitOnFields = (urlObj) => omit(urlObj, URLlistFieldFilter);
const filteredURLObjectList = URLlist.filter(includesType).map(omitOnFields);
```

Reply:
This may be a just-me thing. Please feel free to ignore it.

Reply:
I tend to prefer explicitness over terseness, particularly if it comes up in a review.  I'll add that once I'm on master, etc.